### PR TITLE
user-details-hook

### PR DESCRIPTION
**Pull request by rohan-1898**

### CODE REVIEW

**Comment by shanu798** [comment]
review done

File: src/hooks/useUserDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+type Props = {
+  children : React.ReactNode;
+}
+
+interface UseUserDetailsType {
+  loadingStatus: LOADING_STATUS,
+  userDetails: Record<string, string> | string[],
+  getUserDetails: () => void,
+  verificationOptions: {},
```
Comment (vaibhavg15) [inline src/hooks/useUserDetails.tsx:19]: wrong type.

File: src/hooks/useUserDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  getVerificationOptions: () => void,
+  selectedOptions: Record<string, string>,
+  setFormData: (selectedOptions: Record<string, string>) => void,
+  onSubmit: () => Promise<any>,
+  submitStatus: {},
```
Comment (vaibhavg15) [inline src/hooks/useUserDetails.tsx:24]: wrong type.

File: src/hooks/useUserDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            'ERROR: Failed to fetch user details: ', e,
+          );
+          setLoadingStatus(LOADING_STATUS.FAILED);
+        });
+    }, [userDetails],
+  );
+
+  const getVerificationOptions = useCallback(
+    () => {
+      if (isLoadingOrCompletedOrFailed(loadingStatus)) {
```
Comment (vaibhavg15) [inline src/hooks/useUserDetails.tsx:72]: should have different loading statuses for getverification and getuser details.

File: src/hooks/useUserDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+          );
+          setLoadingStatus(LOADING_STATUS.FAILED);
+        });
+    }, [verificationOptions],
+  );
+
+  const setFormData = useCallback(
+    (newSelectedOptions: Record<string, string>) => {
+      setSelectedOptions({
+        ...selectedOptions, newSelectedOptions,
```
Comment (vaibhavg15) [inline src/hooks/useUserDetails.tsx:93]: newSelectedOptions should also have the spread operator.

File: src/hooks/useUserDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+      });
+    }, [selectedOptions],
+  );
+
+  const onSubmit = useCallback(
+    () => new Promise((
+      resolve, reject,
+    ) => {
+      if (isLoadingOrCompletedOrFailed(loadingStatus)) {
```
Comment (vaibhavg15) [inline src/hooks/useUserDetails.tsx:102]: separate loading status for submit as well

File: src/service/apiUrls.ts [hunk 6]
```diff
@@ -2,6 +2,8 @@ import { getUserDetailsMockData } from './mock-data/getUserDetailsMockData';
 
 export const API_URLS = {
   USER_DETAILS: '/api/UPIN/CLIENT_CODE/user-details',
```
Comment (vaibhavg15) [inline src/service/apiUrls.ts:4]: USER_DETAILS can be an object with all the URLs nested inside that object.

File: src/hooks/useUserDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+} from '@evive/ui-kit';
+import { API_URLS } from '../service/apiUrls';
+
+type Props = {
+  children : React.ReactNode;
+}
+
+interface UseUserDetailsType {
+  loadingStatus: LOADING_STATUS,
+  userDetails: Record<string, string> | string[],
```
Comment (vaibhavg15) [inline src/hooks/useUserDetails.tsx:17]: is the type correct?
can it be an array of strings? it would always be an object, right?

**Review by vaibhavg15 — CHANGES_REQUESTED** [review]

**Pull request merged**